### PR TITLE
Changes for PE2015.3.0

### PR DESCRIPTION
**Pull request by npwalker**
I've been working on this for a while and I want to put it up for PR to get feedback before merging it in.  

Some things to consider / discuss:

1. Since file sync will sync the contents of `/etc/puppetlabs/code/hiera.yaml` is the correct thing to do move hiera.yaml into the control-repo and manage hiera_config in puppet.conf to point to it?  Or should we stick with what I have here which is to just move to `/etc/puppetlabs/puppet/hiera.yaml`?

2. How does one uninstall or at least disable zack/r10k?  
 - https://github.com/acidprime/r10k/issues/255
 - I implemented a way to disable the zack/r10k webhook
    - https://github.com/npwalker/control-repo/commit/ec7a8d81a65051503e9f27e142f29b187327959b

3.  Switching from zack r10k to code manager will leave a webhook setup in the GMS of choice.  This is because the webhooks are unique based on url not name.  I could attempt to clean this up by carrying over the random username and password code and doing an ensure => absent on previous webhook url.  

4.  Switching to code manager requires running puppet 2-3 times before everything is up and working.  I'm not sure there's anyway to avoid this given the constraints but I'm open to ideas.  

5.  The use of zack/r10k webhook or code manager is determined by the version of PE. 
 - https://github.com/npwalker/control-repo/commit/d2db2750284dfeee15f375ce06bbcbc301738b84

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,34 +132,61 @@ http://docs.puppetlabs.com/pe/latest/install_basic.html
...
-     - `{"provider": "rugged",
-    "private_key": "/root/.ssh/r10k_rsa"}`
-3. Run `puppet agent -t` 
+ - `cat /etc/puppetlabs/puppetserver/code_manager.key.pub`
+3. Login to the PE console
+4. Select Access Control in the left hand panel
+5. On the User Roles page, add a new role called `Deploy Environments`
+ - NOTE: Make sure to name it exactly as I have because the puppet code expects that exact name
+6. After creating the role click through and select the permissions tab
+ - Add Puppet Environmnet type, Deploy Code permission, and All object
```
Comment (binford2k) [inline README.md]: environment

File: README.md [hunk 2]
```diff
@@ -109,34 +132,61 @@ http://docs.puppetlabs.com/pe/latest/install_basic.html
 
 ###Get the Control-Repo Deployed On Your Master
 
-At this point you have my control-repo code deployed into your git server.  However, we have one final challenge getting that code onto your puppet master.  In the end state the master will pull code from the git server via r10k, however, at this moment your puppet master doesn't have credentials to get code from the git server.  
+At this point you have my control-repo code deployed into your git server.  However, we have one final challenge getting that code onto your puppet master.  In the end state the master will pull code from the git server via r10k, however, at this moment your puppet master doesn't have credentials to get code from the git server.
```
Comment (binford2k) [inline README.md]: Missing punctuation, "However, we have one final challenge: getting that code onto your puppet master"

File: manifests/site.pp [hunk 3]
```diff
@@ -11,20 +11,8 @@
...
-filebucket { 'main':
-  #server should point to one master that will be the file bucket
-  server => "${settings::server}",
-  path   => false,
-}
-
-# Make filebucket 'main' the default backup location for all File resources:
-File { backup => 'main' }
+# Disable filebucket by default for all File resources:
+File { backup => false }
```
Comment (binford2k) [inline manifests/site.pp]: Are we recommending all clients give up on filebucket? I would agree with that, but how are we messaging that? Will this end up with people expecting and foolishly depending on the backup?

Comment (npwalker) [inline manifests/site.pp]: http://docs.puppetlabs.com/pe/latest/release_notes.html#filebucket-resource-no-longer-created-by-default

Comment (binford2k) [inline manifests/site.pp]: :smile:

**Comment by npwalker** [comment]
All right I think I got a +1 from @glarizza.  Unless I hear a reason not to merge this by EOD tomorrow I'm going to hit the big green button.

**Comment by glarizza** [comment]
Just a comment on 1 above:  I usually put hiera.yaml into the control repo and then have a resource to drop it from the production environment to the confdir, the reason is because, to me, A.) the control repo should manage EVERYTHING necessary to get a Puppet master up and running, B.) It's easier to read/manage in file format versus using the Hiera module (in my opinion and the customers I've polled) -  it's so infrequently changed that having the file there versus a class declaration makes it easier to "find", and C.) hiera_config is a singleton configuration item and is only read once on Puppetserver restart, so because we can't have different copies of it anyways, the easiest thing to do is just ensure it present based on the control repo (and, in the future, if we CAN have a per-environment Hiera config, we're setup for that).

**Comment by npwalker** [comment]
@glarizza Can we open that as a new issue?  I think it's worth discussing and probably changing since I don't really like how it takes a puppet run to update hiera in the current format.  A few people have mentioned this to me and I think I misunderstood Zack when he told me to use hunner/hiera ( as I think he meant for boot strapping as opposed to long term management).  

However, I don't want to hold up this PR because I've been using hunner/hiera for quite a while so this isn't a change in functionality.  

Also, what does "have a resource to drop it from the production environment to the confdir" mean?

**Comment by glarizza** [comment]
@npwalker A file resource to grab the source from /etc/puppetlabs/code/environments/production/hiera.yaml and ensure a file at /etc/puppetlabs/puppet/hiera.yaml, then it bounces the puppetserver service.

**Comment by glarizza** [comment]
And yeah, this shouldn't hold up the PR, just responding to the question I never answered :)

**Pull request merged**